### PR TITLE
Machine ID: Spacelift docs

### LINKED ISSUE #34391: Machine ID: Document Spacelift join
## Details

Add a page explaining how to use Machine ID with Spacelift and the Teleport Terraform Provider

Depends on https://github.com/gravitational/teleport/issues/34389
Depends on https://github.com/gravitational/teleport/issues/34390

## How will we know this is resolved?

Try deploying a simple resource using Teleport's Terraform Provider in Spacelift and using the `spacelift` join method.

### PR DESCRIPTION
**Pull request by strideynet**
Closes https://github.com/gravitational/teleport/issues/34391

Guide for Spacelift and Machine ID. We've moved the existing guide to the Machine ID section to present a single "golden path" rather than two possible implementations.

We'll need to wait for the first release that includes the spacelift runner docker image before we merge any backports - but I don't see the harm in this going into the master docs.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

**Review by ptgott — APPROVED** [review]

File: docs/config.json [hunk 1]
```diff
@@ -1411,13 +1407,17 @@
               "title": "Jenkins",
               "slug": "/machine-id/deployment/jenkins/"
             },
-	    {
+	        {
```
Comment (zmb3) [inline docs/config.json:1410]: Indentation change here?

File: docs/pages/machine-id/deployment/spacelift.mdx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+resources. 
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- A GitHub repository where you will store your Terraform configuration and a
+  Spacelift stack linked to this repository.
+- A paid Spacelift account. This is required to use the `spacelift` join method.
```
Comment (zmb3) [inline docs/pages/machine-id/deployment/spacelift.mdx:30]: Should we make Spacelift joining enterprise only?

Similar to what we did with GitHub Enterprise - if a Teleport integration requires paid 3rd-party software, the Teleport feature itself should probably be for paid customers too.

WDYT?

File: docs/pages/machine-id/deployment/spacelift.mdx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+```yaml
+version: "1"
+stack_defaults:
+  runner_image: public.ecr.aws/gravitational/teleport-spacelift-runner:(=teleport.version=)
+  before_init:
+  - |
+    tbot start --oneshot \
+      --data-dir=memory:// \
+      --auth-server teleport.example.com:443 \
+      --join-method spacelift --token example-bot \
```
Comment (zmb3) [inline docs/pages/machine-id/deployment/spacelift.mdx:185]: nit: put `--token` on its own line so we have one flag per line. On first read I missed this as I quickly assumed this whole line was for the `--join-method`.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/machine-id/deployment/spacelift.mdx (hunk 3 above)
Comment (strideynet) [reply docs/pages/machine-id/deployment/spacelift.mdx]: [7d242e4](https://github.com/gravitational/teleport/pull/34935/commits/7d242e4ce8d207ec63c853c485be87f2635b9274)

File: docs/config.json (hunk 1 above)
Comment (strideynet) [reply docs/config.json]: [7d242e4](https://github.com/gravitational/teleport/pull/34935/commits/7d242e4ce8d207ec63c853c485be87f2635b9274)

**Pull request merged**